tools/esdlcmd/esdl-publish.cpp

```diff
@@ -1095,6 +1095,296 @@ class EsdlUnBindMethodCmd : public EsdlBindMethodCmd
     }
 };
 
+class EsdlBindLogTransformCmd : public EsdlBindServiceCmd
+{
+protected:
+    StringAttr optLogTransform;
+    StringAttr optBindingId;
+
+public:
+    int processCMD()
+    {
+        Owned<IClientWsESDLConfig> esdlConfigClient = EsdlCmdHelper::getWsESDLConfigSoapService(optWSProcAddress, optWSProcPort, optUser, optPass);
+        Owned<IClientConfigureESDLBindingLogTransformRequest> request = esdlConfigClient->createConfigureESDLBindingLogTransformRequest();
+
+        fprintf(stdout,"\nAttempting to configure LogTransform : '%s' for binding '%s'\n", optLogTransform.get(), optBindingId.get());
+        request->setEsdlBindingId(optBindingId.get());
+        request->setLogTransformName(optLogTransform.get());
+        request->setConfig(optInput);
+        request->setOverwrite(optOverWrite);
+
+        if (optVerbose)
+            fprintf(stdout,"\nLogTransform config: %s\n", optInput.get());
+
+        Owned<IClientConfigureESDLBindingLogTransformResponse> resp = esdlConfigClient->ConfigureESDLBindingLogTransform(request);
+
+        if (resp->getExceptions().ordinality()>0)
+        {
+            EsdlCmdHelper::outputMultiExceptions(resp->getExceptions());
+            return 1;
+        }
+
+        outputWsStatus(resp->getStatus().getCode(), resp->getStatus().getDescription());
+
+        return 0;
+    }
+
+    void usage()
+    {
+        printf( "\nUsage:\n\n"
+                "esdl bind-log-transform <TargetESDLBindingID> <TargetLogTransformName> [command options]\n\n"
```

Comment:
Looks like <TargetLogTransformName> is optional and command option --config is required, Actually, a transform name is part of the transform definition referenced by the --config parameter and <TargetLogTransformName> might not be needed at all.

Command option --overwrite is optional.

Reply:
Will change --config is required. But, the transform name is needed because > 1 transforms may be in the  --config. This command only binds 1.